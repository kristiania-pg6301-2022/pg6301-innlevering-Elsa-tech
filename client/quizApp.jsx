import * as React from "react";
import { BrowserRouter, Link, Route, Routes } from "react-router-dom";

export function FrontPage() {
  return (
    <div>
      <h1>Quiz Application</h1>
    </div>
  );
}

function ShowQuestion() {
  return (
    <div>
      <h1>Question</h1>
    </div>
  );
}

export function Application() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path={"/"} element={<FrontPage />} />
        <Route path={"/question"} element={<ShowQuestion />} />
      </Routes>
    </BrowserRouter>
  );
}
