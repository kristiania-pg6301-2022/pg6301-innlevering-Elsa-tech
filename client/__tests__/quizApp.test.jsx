import * as React from "react";
import * as ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import { FrontPage, ShowQuestion, QuestionContext } from "../quizApp.jsx";

describe("Quiz Application", () => {
  it("Shows Frontpage", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FrontPage />
      </MemoryRouter>,
      element
    );
    expect(element.querySelector("h1").innerHTML).toEqual("Quiz Application");
    expect(element.innerHTML).toMatchSnapshot();
  });

  it("Shows question with answer-options", () => {
    const question = {
      id: "1",
      question: "Does this function work?",
      answers: {
        answer_a: "yes",
        answer_b: "no",
        answer_c: "maybe",
      },
    };
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter initialEntries={"/question"}>
        <QuestionContext.Provider value={{ randomQuestion: () => question }}>
          <ShowQuestion />
        </QuestionContext.Provider>
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
  });
});
