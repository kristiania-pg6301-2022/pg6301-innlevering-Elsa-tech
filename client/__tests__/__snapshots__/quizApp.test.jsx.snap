// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Application Shows Frontpage 1`] = `"<div><h1>Quiz Application</h1><ul><li><a href=\\"/question\\">Answer random question</a></li></ul></div>"`;

exports[`Quiz Application Shows front page 1`] = `"<div><h1>Amazing quiz application!</h1><div><a href=\\"/question\\">Take a quiz!</a></div><div><a href=\\"/score\\">See your quiz score here</a></div></div>"`;
